gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/jupiter/core/v4/endpoint/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.core.v4.endpoint.loadbalancer;
+
+import io.gravitee.gateway.jupiter.core.v4.endpoint.ManagedEndpoint;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * @author Guillaume LAMIRAND (guillaume.lamirand at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class RoundRobinLoadBalancer extends AbstractLoadBalancerStrategy {
+
+    private final AtomicInteger counter = new AtomicInteger(0);
+
+    public RoundRobinLoadBalancer(final List<ManagedEndpoint> endpoints) {
+        super(endpoints);
+    }
+
+    @Override
+    public void refresh() {
+        super.refresh();
+        if (counter != null) { //NOSONAR counter could be null if refresh is called from parent constructor
```

Comment:
unfortunately that will lead to duplicate code as they are all doing it.

Reply:
From what I see, the `refresh()` function is called in the constructor of the `AbstractLoadBalancerStrategy`. By default `refresh()` does nothing. The 2 other cases just set the counter to 0 which is useless because the counter is initialized during the instantiation. We could simplify by avoiding calling `refresh()` in the constructor.

Reply:
WeightedRandomLoadBalancer does more than just setting a counter to 0.

Reply:
Yep and it could call refresh itself no ?